### PR TITLE
Retribution cancel 'Chaos Bane' buff rotational option

### PR DESCRIPTION
**Pull request by ncberman**
Creates a new rotational option for Retribution Paladins that allows them to cancel the 'Chaos Bane' buff from Shadowmourne when they reach 10 Soul Fragments.

- Added a boolean to retribution paladin variables
- Created a rotational boolean input in retribution paladin's inputs.ts
- Default value set to false
- Ideally the option only appears when relevant to the user i.e. when Shadowmourne is equipped (currently the user needs to refresh the page after equipping or unequipping Shadowmourne for the option to show/hide)

### CODE REVIEW

**Comment by ncberman** [comment]
Also should mention that there are a lot of dumb commits of me trying to find a solution to model taiaj / SM procs correctly but I ended up removing all those changes and only including the shadowmourne cancel option.

**Comment by vigo2** [comment]
This has already been proposed in https://github.com/wowsims/wotlk/issues/2029. With current modelling for Shadowmourne and Paladin "phantom strikes", cancelling the buff is always a DPS loss. This is highly unlikely to change.

**Comment by ncberman** [comment]
> 

From my understanding that with the sims current modelling of Taiaj and SM that currently that you're correct but after those issues are resolved there is a possibility that Chaos Bane cancelling will be an increase over the buff in single target scenarios. Currently SM is not proccing off of all the paladin abilities that should proc it, taiaj should compound on this as the issue with proc masks is resolved due to it causing more attacks and seal procs.

Old Discussion:
https://www.mmo-champion.com/threads/754663-Ret-Pally-getting-Shadowmourne
At the bottom of the page (Seems dubious but they said they saw a damage increase):
https://wowguidepvpandpve.blogspot.com/2014/10/retribution-high-pve-icc-dps-paladin.html

File: sim/paladin/retribution/rotation.go [hunk 1]
```diff
@@ -316,3 +317,9 @@ func (ret *RetributionPaladin) waitUntilNextEvent(sim *core.Simulation, events [
 
 	sim.AddPendingAction(pa)
 }
+
+func CancelChaosBane(ret *RetributionPaladin, sim *core.Simulation) {
+	if ret.Paladin.HasActiveAura("Chaos Bane") && ret.Paladin.CancelChaosBane {
+		ret.Paladin.GetAura("Chaos Bane").Deactivate(sim)
+	}
```
[the remark this answers is not in the document]
Comment (ncberman) [reply sim/paladin/retribution/rotation.go:327]: Made the change and remerged master into this PR.
ignore some of the extra commits was trying to squash some commits and utterly failed lol.

**Pull request merged**